langchain/tools/tool.py

```diff
@@ -0,0 +1,27 @@
+"""Base implementation for tools or skills."""
+
+import asyncio
+from abc import abstractmethod
+from typing import Any, Awaitable, Callable, Optional
+
+from pydantic import BaseModel
+
+
+class Tool(BaseModel):
```

Comment:
would call basetool and leave previous tool as is rather than renaming dynamic tool

Reply:
ya and maybe also name this file `base.py`, put `BaseToolKit` in here too. Would follow the convention in other directories (i.e. `llms/base.py`, `chains/base.py`)